src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs

```diff
@@ -21,34 +25,78 @@ public static TItem Get<TItem>(this IMemoryCache cache, object key)
 
         public static bool TryGetValue<TItem>(this IMemoryCache cache, object key, out TItem value)
         {
-            object obj = null;
-            if (cache.TryGetValue(key, out obj))
+            object result;
+            if (cache.TryGetValue(key, out result))
             {
-                value = (TItem)obj;
+                value = (TItem)result;
                 return true;
             }
+
             value = default(TItem);
             return false;
         }
 
-        public static object Set(this IMemoryCache cache, object key, object value)
+        public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value)
         {
-            return cache.Set(key, value, new MemoryCacheEntryOptions());
+            return cache.Set(key, value, null);
         }
 
-        public static object Set(this IMemoryCache cache, object key, object value, MemoryCacheEntryOptions options)
+        public static TItem Set<TItem>(this IMemoryCache cache, object key, TItem value, MemoryCacheEntryOptions options)
```

Comment:
MemoryCacheEntryOptions is now a redundant temporary object. Would it make sense to have a few overloads of Set that just took the most common settings and bypassed creating the MemoryCacheEntryOptions? E.g.:
```
Set<TItem>(this IMemoryCache cache, object key, TItem value, DateTimeOffset absoluteExpiration)
Set<TItem>(this IMemoryCache cache, object key, TItem value, TimeSpan absoluteExpirationRelativeToNow)
Set<TItem>(this IMemoryCache cache, object key, TItem value, IChangeToken expirationToken)
```